src/csharp/Microsoft.Spark.Worker/TaskRunner.cs

```diff
@@ -163,9 +165,19 @@ public void Stop()
 
                 // Mark the beginning of the accumulators section of the output
                 SerDe.Write(outputStream, (int)SpecialLengths.END_OF_DATA_SECTION);
+                
+                // Write the accumulator updates this task caused, for consumption by the driver later
+                var accumulators = AccumulatorWorkerRegistry.GetAll().Values.ToList();
+                SerDe.Write(outputStream, accumulators.Count);
+                foreach (var accumulator in accumulators)
+                {
+                    var accumulatorBytes = accumulator.ToByteArray();
+                    SerDe.Write(outputStream, accumulatorBytes.Length);
+                    SerDe.Write(outputStream, accumulatorBytes);
+                }
 
-                // TODO: Extend the following to write accumulator values here.
-                SerDe.Write(outputStream, 0);
+                // We've written all of the accumulator updates, clear out the local cache of accumulators
```

Comment:
Is there a critical zone here where accumulator updates may come in between the fetch and the clear? Or are they all settled by this point?

Reply:
No, once we've reached this point the spark task / job is finished. We've processed all of the rows and we are sending the results of the task, including accumulator updates now, to the spark driver.